pkg/core/resources/apis/meshexternalservice/api/v1alpha1/validator.go

```diff
@@ -1,10 +1,144 @@
 package v1alpha1
 
 import (
+	"fmt"
+	"math"
+	"slices"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+
 	"github.com/kumahq/kuma/pkg/core/validators"
 )
 
+var (
+	allMatchProtocols    = []string{string(TcpProtocol), string(GrpcProtocol), string(HttpProtocol), string(Http2Protocol)}
+	allTlsVersions       = []string{string(TLSVersionAuto), string(TLSVersion10), string(TLSVersion11), string(TLSVersion12), string(TLSVersion13)}
+	allVerificationModes = []string{string(TLSVerificationSkipSAN), string(TLSVerificationSkipCA), string(TLSVerificationSkipAll), string(TLSVerificationSecured)}
+	allSANMatchTypes     = []string{string(SANMatchPrefix), string(SANMatchExact)}
+)
+
 func (r *MeshExternalServiceResource) validate() error {
 	var verr validators.ValidationError
+	path := validators.RootedAt("spec")
+
+	verr.AddErrorAt(path.Field("match"), validateMatch(r.Spec.Match))
+	// when extension != nil then it's up to the extension to validate endpoints and tls
+	if r.Spec.Extension == nil {
+		if r.Spec.Endpoints != nil {
+			verr.AddErrorAt(path.Field("endpoints"), validateEndpoints(r.Spec.Endpoints))
+		}
+
+		if r.Spec.Tls != nil {
+			verr.AddErrorAt(path.Field("tls"), validateTls(r.Spec.Tls))
+		}
+	}
+
+	if r.Spec.Extension != nil && r.Spec.Extension.Type == "" {
+		verr.AddViolationAt(path.Field("extension").Field("type"), validators.MustNotBeEmpty)
+	}
+
 	return verr.OrNil()
 }
+
+func validateTls(tls *Tls) validators.ValidationError {
+	var verr validators.ValidationError
+
+	if tls.Version != nil {
+		path := validators.RootedAt("version")
+		if tls.Version.Min != nil {
```

Comment:
Should we validate cases:
- min is Auto and max is specific
- max is Auto and min is specific
- min is higher than max